Cannot build python38packages.pycairo on unstable

### Steps To Reproduce
Steps to reproduce the behavior:
1. nix build ".#python38Packages.pycairo"

### Build log
```
error: builder for '/nix/store/ndgm8jqasbassscs48zd5fh2d4a86x83-python3.8-pycairo-1.21.0.drv' failed with exit code 1;
       last 10 log lines:
       > compilation terminated.
       > [16/21] Compiling C object cairo/_cairo.cpython-38-x86_64-linux-gnu.so.p/bufferproxy.c.o
       > FAILED: cairo/_cairo.cpython-38-x86_64-linux-gnu.so.p/bufferproxy.c.o
       > gcc -Icairo/_cairo.cpython-38-x86_64-linux-gnu.so.p -Icairo -I../cairo -I/nix/store/vc95a7ahizgj62q8kfbrfhah7c8b481b-python3-3.8.15/include/python3.8 -I/nix/store/3x7jx65ygxzbkq226n8vd3b1jii6a23x-cairo-1.16.0-dev/include/cairo -I/nix/store/y2qjg1g3ngwmm28ys05g7pkk0nw2jxgh-freetype-2.12.1-dev/include/freetype2 -I/nix/store/y2qjg1g3ngwmm28ys05g7pkk0nw2jxgh-freetype-2.12.1-dev/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -fPIC -DPYCAIRO_VERSION_MAJOR=1 -DPYCAIRO_VERSION_MINOR=21 -DPYCAIRO_VERSION_MICRO=0 -Wall -Warray-bounds -Wcast-align -Wconversion -Wextra -Wformat=2 -Wformat-nonliteral -Wformat-security -Wimplicit-function-declaration -Winit-self -Winline -Wmissing-format-attribute -Wmissing-noreturn -Wnested-externs -Wold-style-definition -Wpacked -Wpointer-arith -Wreturn-type -Wshadow -Wsign-compare -Wstrict-aliasing -Wundef -Wunused-but-set-variable -Wswitch-default -Wno-missing-field-initializers -Wno-unused-parameter -fno-strict-aliasing -fvisibility=hidden -MD -MQ cairo/_cairo.cpython-38-x86_64-linux-gnu.so.p/bufferproxy.c.o -MF cairo/_cairo.cpython-38-x86_64-linux-gnu.so.p/bufferproxy.c.o.d -o cairo/_cairo.cpython-38-x86_64-linux-gnu.so.p/bufferproxy.c.o -c ../cairo/bufferproxy.c
       > In file included from ../cairo/bufferproxy.c:32:
       > /nix/store/vc95a7ahizgj62q8kfbrfhah7c8b481b-python3-3.8.15/include/python3.8/Python.h:44:10: fatal error: crypt.h: No such file or directory
       >    44 | #include <crypt.h>
       >       |          ^~~~~~~~~
       > compilation terminated.
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/ndgm8jqasbassscs48zd5fh2d4a86x83-python3.8-pycairo-1.21.0.drv'.
```

### Notify maintainers
There is not maintainer. You made the last commit @mweinelt.

### Metadata
 - system: `"x86_64-linux"`
 - host os: `Linux 6.0.0, NixOS, 22.11 (Raccoon), 22.11.20221008.c592415`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(sweenu): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

--- a/pkgs/development/python-modules/pycairo/default.nix
+++ b/pkgs/development/python-modules/pycairo/default.nix
@@ -7,6 +7,7 @@
 , pytestCheckHook
 , pkg-config
 , cairo
+, libxcrypt
 , python
 }:
 
@@ -33,6 +34,8 @@ buildPythonPackage rec {
 
   buildInputs = [
     cairo
+  ] ++ lib.optionals (pythonOlder "3.9") [
+    libxcrypt
   ];
 
   checkInputs = [

